src/Workspaces/CSharpTest/Formatting/FormattingTests.cs

```diff
@@ -4819,6 +4819,363 @@ void bar()
+        2,
+        3);
+    }
+}";
+            // PROTOTYPE: missing indent on 2 and 3
+            await AssertFormatAsync(expectedCode, code);
+        }
+
+        [Fact]
+        [Trait(Traits.Feature, Traits.Features.Formatting)]
+        public async Task FormatRecursivePattern_Positional_Multiline2()
+        {
+            var code = @"class C
+{
+    void M() {
+_ = this is  ( 1 ,
+2 ,
+3 )  ; }
+}";
+            var expectedCode = @"class C
+{
+    void M()
+    {
+        _ = this is (1,
+        2,
+        3);
+    }
+}";
+            // PROTOTYPE: missing indent on 2 and 3
+            await AssertFormatAsync(expectedCode, code);
+        }
+
+        [Fact]
+        [Trait(Traits.Feature, Traits.Features.Formatting)]
+        public async Task FormatRecursivePattern_Positional_Multiline3()
+        {
+            var code = @"class C
+{
+    void M() {
+_ = this is
+( 1 ,
+2 ,
+3 )  ; }
+}";
+            var expectedCode = @"class C
+{
+    void M()
+    {
+        _ = this is
+        (1,
+        2,
+        3);
+    }
+}";
+            // PROTOTYPE: missing indent on open paren, 2 and 3
+            await AssertFormatAsync(expectedCode, code);
+        }
+
+        [Fact]
+        [Trait(Traits.Feature, Traits.Features.Formatting)]
+        public async Task FormatRecursivePattern_Positional_Multiline4()
+        {
+            var code = @"class C
+{
+    void M() {
+_ = this is
+( 1 ,
+2 , 3 )  ; }
+}";
+            var expectedCode = @"class C
+{
+    void M()
+    {
+        _ = this is
+        (1,
+        2, 3);
+    }
+}";
+            // PROTOTYPE: missing indent on open paren, 2 and 3
+            await AssertFormatAsync(expectedCode, code);
+        }
+
+        [Fact]
+        [Trait(Traits.Feature, Traits.Features.Formatting)]
+        public async Task FormatRecursivePattern_Properties_Singleline()
+        {
+            var code = @"class C
+{
+    void M() { _ = this is  {  P1 :  1  } ; }
+}";
+            var expectedCode = @"class C
+{
+    void M() { _ = this is { P1: 1 }; }
+}";
+
+            await AssertFormatAsync(expectedCode, code);
+        }
+
+        [Fact]
+        [Trait(Traits.Feature, Traits.Features.Formatting)]
+        public async Task FormatRecursivePattern_Properties_Multiline()
+        {
+            var code = @"class C
+{
+    void M() {
+_ = this is
+{
+P1 :  1  ,
+P2 : 2
+} ;
+}
+}";
+            var expectedCode = @"class C
+{
+    void M()
+    {
+        _ = this is
+        {
+        P1: 1,
+        P2: 2
+        };
+    }
+}";
+
+            await AssertFormatAsync(expectedCode, code);
+        }
+
+        [Fact]
+        [Trait(Traits.Feature, Traits.Features.Formatting)]
+        public async Task FormatRecursivePattern_Properties_Multiline2()
+        {
+            var code = @"class C
+{
+    void M() {
+_ = this is {
+P1 :  1  ,
+P2 : 2
+} ;
+}
+}";
+            var expectedCode = @"class C
+{
+    void M()
+    {
+        _ = this is {
+        P1: 1,
+        P2: 2
```

Comment:
indentation seems wrong here.

Reply:
I've [tried various things](https://github.com/dotnet/roslyn/pull/26474/commits/3280d92eb6460d68318c3b0ec53eb0babc3db088) in `IndentBlockFormattingRules`, but none seems to have any effect.

@CyrusNajmabadi @heejaechang Any tips on this?

Reply:
let me see.

Reply:
this kind of thing (expression level), we do not force indentation. looks like it is doing write thing. it is maintaining relative indentation from "_"

if test was written
_ = this is {
    P1 : 1 ...

it will maintain relative indentation between _ and P1.

you can have method call
method(1,
2, 
     4);

and format, you will see  relative indentation of 2 and 4 are maintained aginst method

Reply:
The latest commit I pushed finally has some effect. But it's super weird. Only the first arm of the switch expression got indented... I'll continue poking around.